pkg/scheduler/framework/plugins/volumebinding/binder.go

```diff
@@ -356,6 +366,40 @@ func (b *volumeBinder) FindPodVolumes(pod *v1.Pod, boundClaims, claimsToBind []*
 	return
 }
 
+// GetEligibleNodes returns a list of eligible nodes by name based on the bound claims of the pod to
+// potentially reduce down the list of nodes in subsequent scheduling stages
+func (b *volumeBinder) GetEligibleNodes(boundClaims []*v1.PersistentVolumeClaim) (sets.String, error) {
+	if len(boundClaims) == 0 {
+		return sets.NewString(), nil
+	}
+
+	var errs []error
+	eligibleNodes := sets.NewString()
+	for _, pvc := range boundClaims {
+		pvName := pvc.Spec.VolumeName
+		pv, err := b.pvCache.GetPV(pvName)
+		if err != nil {
+			if _, ok := err.(*errNotFound); ok {
+				err = nil
+			}
+			errs = append(errs, err)
+			continue
+		}
+
+		// if the PersistentVolume is a local PV, then add the node to the eligible nodes
+		if nodeName, isLocalPV := util.GetLocalLocalPersistentVolumeNodeName(pv); isLocalPV {
+			eligibleNodes.Insert(nodeName)
+		}
+	}
+
+	klog.V(4).InfoS("GetEligibleNodes: reduced down eligible nodes", "nodes", eligibleNodes, "error", errs)
+
+	if len(errs) > 0 {
+		return sets.NewString(), errors.NewAggregate(errs)
+	}
```

Comment:
it means you're returning an empty string once there is an error, so why not return earlier when the first error is caught?

Reply:
There is a logging line above and I think it would be helpful to log all the errors encountered for all Bound claims rather than see an issue and immediately return.